### PR TITLE
chore(issues-stream-assignees): Deprecate old assignee selector component 

### PR DESCRIPTION
**Pull request by MichaelSun48**
Appends "Deprecated" to all instances of `<AssigneeSelector>` and `<AssigneeSelectorDropdown>` in preparation for new component and adds @deprecated tag. 

There will be one component that replaces both of these that is also called `<AssigneeSelectorDropdown>`. The PR to add this component has not been merged yet, so the deprecated tag will be temporarily misleading.

### CODE REVIEW

**Review by michellewzhang — COMMENTED** [review]
just curious, why make this a separate PR instead of making the new component changes directly?

**Comment by MichaelSun48** [comment]
> just curious, why make this a separate PR instead of making the new component changes directly?

@michellewzhang 
Just for the sake of limiting the scope of [that PR](https://github.com/getsentry/sentry/pull/69639) and making it easier to review. It is currently WIP and is already +500 lines of code

**Review by michellewzhang — APPROVED** [review]
lgtm!

File: static/app/components/deprecatedAssigneeSelector.tsx [hunk 1]
```diff
@@ -21,9 +21,12 @@ import type {Actor} from 'sentry/types/core';
 import type {SuggestedOwnerReason} from 'sentry/types/group';
 import useOrganization from 'sentry/utils/useOrganization';
 
-interface AssigneeSelectorProps
+/**
+ * @deprecated use AssigneeSelectorDropdown instead (Coming in future PR)
```
Comment (malwilley) [inline static/app/components/deprecatedAssigneeSelector.tsx:25]: nit: no need to put this here, only really needed on the component

**Review by malwilley — APPROVED** [review]

**Pull request merged**